extensions/ql-vscode/src/databases/config/db-config-store.ts

```diff
@@ -144,6 +154,32 @@ export class DbConfigStore extends DisposableObject {
     );
   }
 
+  public doesRemoteDbExist(dbName: string, listName?: string): boolean {
+    if (!this.config) {
+      throw Error(
+        "Cannot check remote database existence if config is not loaded",
+      );
+    }
+
+    if (listName) {
+      return this.config.databases.remote.repositoryLists.some(
+        (l) => l.name === listName && l.repositories.includes(dbName),
+      );
+    }
```

Comment:
Optional: I'm wondering if we can solve this in a more general way:

```
this.configErrors = this.configValidator.validate(config);
if (this.configErrors.length > 0) {
  this.configErrors.forEach((error) => {
    throw Error(error.message);
  });
}
```

and use it for alle checks? Our error messages would be a bit more general - I checked for adding a new repo and it looks good (albeit we could add an 'Error: ' at the beginning of the message):

<img width="467" alt="image" src="https://user-images.githubusercontent.com/29482385/208876435-1a37c683-b93d-4a7b-bb60-d63f5babafd6.png">

I think it would clean up the file and make it easier to read as well as make sure we catch all errors.

Reply:
Thanks for thinking about readability! 

We don't want to throw an exception in this case, we just want to check if something already exists and make a decision in the UI based on that. I think the fact that I'm throwing an exception further up is confusing (and wrong) so I'll change it as I mention in https://github.com/github/vscode-codeql/pull/1893/files#r1054201240.

Essentially, I don't think we should have unhandled exceptions for things the user has done "wrong".

Reply:
🤔 Hm then I'm misunderstanding something, as we're throwing errors everywhere in this file and they are shown in the UI..

Reply:
The db-config-store adds checks and throws error to avoid the config becoming invalid.

However, in reality we _should_ never reach those errors. The UI should never try to add a new list if it already exists. It should first check if it exists and then either show an error to the user or add the list. Note that I say _should_ never because a) we could introduce bugs in our UI code, so this is another line of defence and b) timing could in theory cause this (e.g. if a user does some UI actions + edits the db config file directly, at the same time)

Reply:
👍 Ok, sounds good, thanks for the explanation!